test/extra/NotificationTest.cpp

```diff
@@ -9,29 +9,122 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include <google/protobuf/util/message_differencer.h>
 #include <gtest/gtest.h>
+#include <up-cpp/communication/NotificationSink.h>
+#include <up-cpp/communication/NotificationSource.h>
+
+#include "UTransportMock.h"
+#include "up-cpp/datamodel/serializer/UUri.h"
 
 namespace {
+using namespace uprotocol::communication;
+using namespace uprotocol::datamodel::serializer::uri;
+using MsgDiff = google::protobuf::util::MessageDifferencer;
 
-class TestFixture : public testing::Test {
+class NotificationTest : public testing::Test {
 protected:
 	// Run once per TEST_F.
 	// Used to set up clean environments per test.
-	void SetUp() override {}
+	void SetUp() override;
 	void TearDown() override {}
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	NotificationTest() = default;
+	~NotificationTest() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
+
+	void buildValidTestTopic();
+	void buildValidDefaultSourceURI();
+	uprotocol::v1::UUri testTopic_;
+	uprotocol::v1::UUri testDefaultSourceURI_;
+	uprotocol::v1::UPayloadFormat format_;
+	std::optional<uprotocol::v1::UPriority> priority_;
+	size_t capture_count_ = 0;
+	uprotocol::v1::UMessage capture_msg_;
+	std::optional<std::chrono::milliseconds> ttl_;
+
+public:
+	void handleCallbackMessage(const uprotocol::v1::UMessage& message);
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+void NotificationTest::SetUp() {
+	buildValidDefaultSourceURI();
+	buildValidTestTopic();
+}
+
+void NotificationTest::buildValidDefaultSourceURI() {
+	testDefaultSourceURI_.set_authority_name("10.0.0.1");
+	testDefaultSourceURI_.set_ue_id(0x18000);
+	testDefaultSourceURI_.set_ue_version_major(0x1);
+	testDefaultSourceURI_.set_resource_id(0x0);
+
+	format_ = uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+	priority_ = uprotocol::v1::UPriority::UPRIORITY_CS1;
+	ttl_ = std::chrono::milliseconds(1000);
+}
+
+void NotificationTest::buildValidTestTopic() {
+	testTopic_.set_authority_name("10.0.0.2");
+	testTopic_.set_ue_id(0x00011101);
+	testTopic_.set_ue_version_major(0xF8);
+	testTopic_.set_resource_id(0x8101);
+}
+
+void NotificationTest::handleCallbackMessage(
+    const uprotocol::v1::UMessage& message) {
+	std::cout << "NotificationTest::handleCallbackMessage\n";
+	capture_msg_ = message;
+	capture_count_++;
+}
+
+TEST_F(NotificationTest, NotificationSuccess) {
+	// Notify Sink
+	auto transportMockNotificationSink =
+	    std::make_shared<uprotocol::test::UTransportMock>(
+	        testDefaultSourceURI_);
+
+	auto callback = [this](const auto& arg) {
+		return this->handleCallbackMessage(arg);
+	};
```

Comment:
I would recommend following along with the updates to the pub/sub version of this test.

In this instance, there is no need for this extra layer of indirection:
```suggestion
	uprotocol::v1::UMessage captured_message;
	auto callback = [&captured_message](const auto& message) {
		captured_message = message;
	};
```

Reply:
Done